File: ProductExtractor/repository/category_repository.py
from repository.utils import insert, select
from models.category import Category
from repository.repository import Repository


class CategoryRepository(Repository[Category]):
    @classmethod
    def tablename(cls) -> str:
        return "Category"

    @classmethod
    def add(cls, item: Category) -> int:
        rows = select(cls.tablename(), ["ID"], {"Name": item.name})

        if len(rows) > 0:
            return rows[0][0]

        return insert(
            cls.tablename(),
            {
                "Name": item.name,
            },
        )

File: src/Models/CategoryRepository.php
<?php

namespace App\Models;
use Exception;

class CategoryRepository extends Repository {
    public static string $tableName = "ProductOffer";

    public static function getByID(int $categoryID): array {
        $data = self::select(['categoryID' => $categoryID]);
        return array_map(self::convertToCategory(...), $data);
    }

    private static function convertToCategory(object $data): Category {
        if (!$data) {
            throw new Exception("unable to convert data into ProductOffer");
        };

        return new Category(
            $data->ID,
            $data->Name,
        );
    }
}
